### PR TITLE
Add max-age ServerCacheControl.DISABLED/REVALIDATE

### LINKED ISSUE #4290: ServerCacheControl does not specify max-age for REVALIDATE/DISABLED
After a short discussion about `Cache-Control` and the parameter `must-revalidate` in the server, it was agreed that the parameter `max-age=0` should be added to be extra correct as the `must-revalidate` already present on `ServerCacheControl.REVALIDATE` and `ServerCacheControl.DISABLED` works on "stale", and "stale" is decided by max-age. 

I cannot find any documentation on what the default value for "stale" is when `must-revalidate` is used without `max-age`. It may be that for those proxies/caches that understand & respect `no-cache`, that they let `no-cache` take precedence.

Link to discussion:
https://line-armeria.slack.com/archives/C1NGPBUH2/p1654675187733719

More information about the header parameter
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#up-to-date_contents_always
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
https://web.dev/http-cache/

### PR DESCRIPTION
**Pull request by tobias-**
Motivation:

According to `Cache-Control` spec, `must-revalidate` operates together with "stale" state, and an easy (the only?) way to specify "stale" is by adding `max-age=0` to the `Cache-Control` response header.

See #4290 for more information.

Modifications:

- Added `max-age=0` to the `ServerCacheControl.DISABLED` and `ServerCacheControl.REVALIDATE` constants

Result:

- Closes #4290
- Should only have benign effect.
  Theoretically a user of Cache-Control could've used e.g. `ServerCacheControl.DISABLED.toBuilder().mustRevalidate(false).build()` and get unexpected result, but I doubt it.

### CODE REVIEW

**Review by trustin — APPROVED** [review]
Thanks a lot for fixing this, @tobias- 🙇

File: core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java [hunk 1]
```diff
@@ -45,6 +45,7 @@ public final class ServerCacheControl extends CacheControl {
      * {@code "no-cache, no-store, must-revalidate"}.
```
Comment (jrhee17) [inline core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java:45]: super minor nit, could also update the documentation

```suggestion
     * {@code "no-cache, no-store, must-revalidate, max-age=0"}.
```

**Review by jrhee17 — APPROVED** [review]
Thanks @tobias- 👍 🙇 👍

File: core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java (hunk 1 above)
Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java:45]: Should `max-age=0` be next to `no-cache` based on the building order?
```suggestion
     * {@code "no-cache, max-age=0, no-store, must-revalidate"}.
```

File: core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java [hunk 2]
```diff
@@ -53,6 +54,7 @@ public final class ServerCacheControl extends CacheControl {
      * {@code "no-cache, must-revalidate"}.
```
[the remark this answers is not in the document]
Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java:54]: Ditto. Add ` max-age=0` next to `no-cache`?

**Review by ikhoon — COMMENTED** [review]
Thanks, left super minor nits for Javadoc.

File: core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java (hunk 1 above)
Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/common/ServerCacheControl.java]: Please ignore my review. I thought they are stringified with the building order.

**Review by ikhoon — APPROVED** [review]
Thanks a lot, @tobias-!

**Review by minwoox — APPROVED** [review]
Thanks! 😄

**Pull request merged**